lib/bson/dbref.rb

```diff
@@ -0,0 +1,130 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+# Copyright (C) 2015-2021 MongoDB Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module BSON
+
+  # Represents a DBRef document in the database.
+  #
+  # @since 2.1.0
+  class DBRef < Document
+    include JSON
+
+    # The constant for the collection reference field.
+    #
+    # @since 2.1.0
+    COLLECTION = '$ref'.freeze
+
+    # The constant for the id field.
+    #
+    # @since 2.1.0
+    ID = '$id'.freeze
+
+    # The constant for the database field.
+    #
+    # @since 2.1.0
+    DATABASE = '$db'.freeze
+
+    # @return [ String ] collection The collection name.
+    def collection
+      self[COLLECTION]
+    end
+
+    # @return [ BSON::ObjectId ] id The referenced document id.
+    def id
+      self[ID]
+    end
+
+    # @return [ String ] database The database name.
+    def database
+      self[DATABASE]
+    end
+
+    # Get the DBRef as a JSON document
+    #
+    # @example Get the DBRef as a JSON hash.
+    #   dbref.as_json
+    #
+    # @return [ Hash ] The max key as a JSON hash.
+    #
+    # @since 2.1.0
+    def as_json(*args)
+      {}.merge(self)
```

Comment:
This should use `update` rather than `merge`.

Reply:
Done. Out of curiosity, why?

Reply:
`merge` is essentially equivalent to `dup.update` - it duplicates the receiver then merges the arguments. Since the receiver is created right here (it's the empty hash) it does not need to be duplicated.